addons/survey/models/survey_question.py

```diff
@@ -93,6 +93,11 @@ def default_get(self, fields):
     matrix_row_ids = fields.One2many(
         'survey.question.answer', 'matrix_question_id', string='Matrix Rows', copy=True,
         help='Labels used for proposed choices: rows of matrix')
+    # -- scoreable/answerable simple answer_types: numerical_box / date / datetime
+    numerical_box_answer = fields.Float('Numerical answer', help="Correct number answer for this question when score is > 0.")
+    date_answer = fields.Date('Date answer', help="Correct date answer for this question when score is > 0.")
+    datetime_answer = fields.Datetime('Datetime answer', help="Correct date and time answer for this question when score is > 0.")
+    answer_score = fields.Float('Score', help="Score value for a correct answer to this question. A score of 0 means no correct answer.")
```

Comment:
Agreed. Added checkbox. Since associated view restricts setting `is_scored_question = True` for question types other than `numerical_box`, `date`, and `datetime`, I simplified some of the scoring logic that was previously looking for those question types. There may be some unexpected behavior (final correct/incorrect/score statistics) in the edge case where someone changes from a question type that can be scored with  `is_scored_question = True` to a question type that cannot be scored, but this way the code is cleaner/more maintainable. If we want to avoid this unexpected behavior I can add in the specific question type checks back in where needed.

Reply:
You could also make the is_scored_question field a compute store=True readonly=False that sets itself to False if the question_type is not withing the acceptable ones.

Reply:
Smart. Added in compute, compute function, store=True, and readonly=False.